dbt/parser.py

```diff
@@ -341,57 +306,61 @@ def parse_schema_tests(tests, root_project, projects):
     return to_return
 
 
-def parse_schema_test(test_base, model_name, test_config, test_type,
-                      root_project_config, package_project_config,
-                      all_projects):
-    if test_type == 'not_null':
-        raw_sql = QUERY_VALIDATE_NOT_NULL.format(
-            ref="{{ref('"+model_name+"')}}", field=test_config)
-        name_key = test_config
+def get_nice_schema_test_name(test_type, test_name, args):
+
+    flat_args = []
+    for arg_name in sorted(args):
+        arg_val = args[arg_name]
+
+        if isinstance(arg_val, dict):
+            parts = arg_val.values()
+        elif isinstance(arg_val, (list, tuple)):
+            parts = arg_val
+        else:
+            parts = [arg_val]
 
-    elif test_type == 'unique':
-        raw_sql = QUERY_VALIDATE_UNIQUE.format(
-            ref="{{ref('"+model_name+"')}}", field=test_config)
-        name_key = test_config
+        flat_args.extend([str(part) for part in parts])
 
-    elif test_type == 'relationships':
-        if not isinstance(test_config, dict):
-            return None
+    clean_flat_args = [re.sub('[^0-9a-zA-Z_]+', '_', arg) for arg in flat_args]
+    unique = "__".join(clean_flat_args)
+    return '{}_{}_{}'.format(test_type, test_name, unique)
 
-        child_field = test_config.get('from')
-        parent_field = test_config.get('field')
-        parent_model = test_config.get('to')
 
-        raw_sql = QUERY_VALIDATE_REFERENTIAL_INTEGRITY.format(
-            child_field=child_field,
-            child_ref="{{ref('"+model_name+"')}}",
-            parent_field=parent_field,
-            parent_ref=("{{ref('"+parent_model+"')}}"))
+def as_kwarg(key, value):
+    test_value = to_string(value)
+    is_function = re.match(r'^\s*(ref|var)\(.+\)$', test_value) is not None
```

Comment:
i feel ok about it. not in love with regex parsing but what can you do.

is the intention to support ref and var in custom schema tests? it doesn't look like those would be passed in here unless i'm missing something